src/classes/CRLP_DefaultConfigBuilder_SVC.cls

```diff
@@ -626,10 +641,108 @@ public class CRLP_DefaultConfigBuilder_SVC {
             groups.add(groupPaymentsWrittenOffContacts);
         }
 
-
         return groups;
     }
 
+    /*************************************************************************************************************
+     * @description Soft deletes all extra Filter Rules in the NPSP Filter Groups upon CRLP Reset
+     * @param List<FilterGroup> the default Filter Groups (wrapper classes) with their lists of default Filter Rules
+     * @return List<FilterRule> defined Filter Rule wrapper classes
+     */
+    private static List<CRLP_RollupCMT.FilterRule> softDeleteExtraFilterRulesInNPSPFilterGroups(List<CRLP_RollupCMT.FilterGroup> groups) {
+
+        List<CRLP_RollupCMT.FilterRule> rulesToSoftDelete = new List<CRLP_RollupCMT.FilterRule>();
+        List<Filter_Rule__mdt> allExistingRules = CMT_FilterRule_SEL.getFilterRules();
+
+        // only proceed if there are existing rules to compare to
+        if (!allExistingRules.isEmpty()) {
+
+            Set<String> defaultRuleNames = new Set<String>();
+            Set<String> defaultGroupNames = new Set<String>();
+            for (CRLP_RollupCMT.FilterGroup grp : groups) {
+                defaultGroupNames.add(grp.recordName);
+                for (CRLP_RollupCMT.FilterRule rule : grp.rules) {
+                    defaultRuleNames.add(rule.recordName);
+                }
+            }
+
+            for (Filter_Rule__mdt rule : allExistingRules) {
+                if (!defaultRuleNames.contains(rule.DeveloperName) && defaultGroupNames.contains(rule.Filter_Group__r.DeveloperName)) {
```

Comment:
Did you consider just checking the start of the developer name for NPSP/NPSP_ instead?

Reply:
Yes but we didn't want to rely on a naming convention, so we went for the (slightly less performant) loop of the actual record names.